traffic_ops/testing/api_contract/v4/test_cdns.py

```diff
@@ -0,0 +1,72 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""Api Contract Test Case for cdns endpoint."""
+import json
+import logging
+import pytest
+
+# Create and configure logger
+logger = logging.getLogger()
+
+
+@pytest.fixture(name="get_cdn_data")
+def get_cdn_prereq_data() -> dict:
+    """PyTest Fixture to store prereq data for cdns endpoint.
+    :returns cdn_data: Returns prerequisite data for cdns endpoint
+    """
+    # Response keys for cdns endpoint
+    with open("prerequisite_data.json", encoding="utf-8", mode="r") as prereq_file:
+        data = json.load(prereq_file)
+    cdn_data = data["cdns"]
+    return cdn_data
+
+
+def test_get_cdn(to_session: object, get_cdn_data: dict, cdn_prereq: list) -> None:
```

Comment:
the type of `to_session` is probably actually `TOSession`. At the very least it can't be `object`, because `object` doesn't have a `delete_cdn_by_id` attribute.

assuming that `get_cdn_data` is the fixture by the same name above, the most specific you can be about its type is that it's an `object` - unless you want to do some checking in that fixture and return a more specific, more concrete type.

assuming that `cdn_prereq` is the same as the fixture by the same name in the `conftest.py` file, the type is known to be more specific than `list` (see above note)

Reply:
Changed the type of to_session to TOSession and also added specific types for both the fixtures.